Cirq for Quantum Computing (#7460)
section-engineering-education/engineering-education: content/articles/cirq-for-quantum-computing/index.md

> We will use Google Colab for this build.  

### Outline
- [An Brief History](#a-brief-history)
- [Wrapping up](#wrapping-up)
- [Further reading](#further-reading)

### A Brief History 
In 2019, Google announced that it had reached a quantum advantage. Quantum advantage is the threshold where a quantum computer can outperform a classical computer for a certain problem. This was a huge achievement in the world of quantum computers.

There are several Google Quantum Hardware that has been used by the Google Research team to date. [Foxtail](https://quantumai.google/cirq/google/devices#foxtail) is one of the hardware devices that was released in 2016, [Bristlecone](https://quantumai.google/cirq/google/devices#bristlecone) is a 72-qubit device and was released in 2018, and most recently, [Sycamore](https://quantumai.google/cirq/google/devices#sycamore) released in 2019. Sycamore is the 54-qubit quantum hardware device that was used to reach the quantum advantage mentioned above. These results are published in the, [Quantum supremacy using a programmable superconducting processor](https://www.nature.com/articles/s41586-019-1666-5) research paper in the Nature journal.  

> Cirq is a quantum computing programming language. It is an open-source, Python framework for writing, manipulating, and optimizing quantum circuits. It can be used to run these programs on simulators and real quantum computers.   

Reviewer: ```suggestion
Cirq is a quantum computing programming language. It is an open-source Python framework for writing, manipulating, and optimizing quantum circuits. It is used to run these programs on simulators and real quantum computers. 
```

### Implementing an example

> To get Cirq installed in your notebook, we'll perform a quick pip install.

Reviewer: ```suggestion
To get Cirq installed in your notebook, we will perform a quick `pip` install.
```

Let's build the circuit by adding these two gates to our circuit.

```python
q_circuit = cirq.Circuit(
    cirq.H(x),
    cirq.CNOT(x, y),
    cirq.measure(x, y)
)
```
- `cirq.measure(x, y)`: We use the `measure()` method and pass the two qubits that we want to measure.
- `cirq.H(x)`: We apply the Hadamard gate onto the `x` qubit.
- `cirq.CNOT(x, y)`: Qubit `x` is the control qubit while qubit `y` is the target qubit.

We can now print the circuit and take a look. We use Python's `print()` method for this task.

```python
print(q_circuit)
```
Output:

![Circuit](/engineering-education/cirq-for-quantum-computing/circuit.jpg)

From the image, we can see the Hadamard gate on qubit `x`, and the `CNOT` gate on qubit `x` and `y`. The `M` on both qubits represents the measurements.

Now that we have built the circuit, let's run it on the simulator. We perform this step by writing the following code:

```python
simulator = cirq.Simulator()
output = simulator.run(q_circuit, repetitions = 20)
print(output)
```

```bash
x,y=11011000000111111111, 11011000000111111111
```

> Lets add an `X` gate to qubit `y` and see whether our result will be different.

Reviewer: ```suggestion
Let's add an `X` gate to qubit `y` and see whether our result will be different.
```
